ql/src/java/org/apache/hadoop/hive/ql/ddl/table/misc/truncate/TruncateTableAnalyzer.java

```diff
@@ -80,8 +86,25 @@ public void analyzeInternal(ASTNode root) throws SemanticException {
     checkTruncateEligibility(root, tableNode, tableNameString, table);
 
     Map<String, String> partitionSpec = getPartSpec((ASTNode) tableNode.getChild(1));
+
     addTruncateTableOutputs(tableNode, table, partitionSpec);
 
+    if (table.getStorageHandler() != null && !table.getStorageHandler().shouldTruncate(table, partitionSpec)) {
```

Comment:
is it `shouldTruncate` or `canTrancate`?

Reply:
Made it to `canTruncate`.

Reply:
@SourabhBadhya have you pushed all your local changes?

Reply:
@deniskuzZ  - https://github.com/apache/hive/pull/4672/files#diff-6dddcdd0457f600011f14e64854f8c030f9d1ae509b1e9fc63061dd0c8596a54R94